### ISSUE TITLE
Clarify "argument has unknown bounds, bounds expected because the <nth> parameter has bounds" message

### ISSUE THREAD

**Issue by AnnaKornfeldSimpson** [microsoft/checkedc-clang#516]
I found the "error: argument has unknown bounds, bounds expected because the 1st parameter has bounds" to be a bit confusing. Besides the message it showed the beginning of the line `memcpy(**dest**, ...` with dest underlined. What am I supposed to fix here? Why is it complaining? I had already given dest bounds (or at least I thought I had), but the other parameters were on separate lines.

**Comment by dtarditi** [comment]
@AnnaKornfeldSimpson, I'm not sure what part of the error message is confusing you.  Could you clarify this?

 In C terminology, an argument is the expression passed to a call.  The parameter is the name of the variable.  I can think of a number of alternative messages:
- The problem is the terminology:  Argument expression has unknown bounds.  Bounds expected because the corresponding parameter has bounds.
-  The problem is the reference to the 1st parameter: Argument has uknown bounds.  Bounds expected because the 1st parameter has bounds."
- The problem is the notes: we should have an error message that points at the argument, along with a note about pointing at the corresponding parameter.

**Comment by AnnaKornfeldSimpson** [comment]
@dtarditi I'm not sure I can entirely re-create my thought process here since I don't have access to whatever code I committed in the neighborhood of July 9 (and I have no idea whether it would be worthwhile for you to search for it). I believe I thought that `dest` (the expression I had passed in as the argument) had bounds (certainly I knew `memcpy` was expecting bounds its parameters) and thus was confused at the error message. We might have to close this one for lack of information, sorry about that!
